From 348a1462aada65ac51c32850e8b94d7f7153f973 Mon Sep 17 00:00:00 2001
From: Josh Wilsdon <jwilsdon@joyent.com>
Date: Tue, 2 Apr 2019 21:46:16 -0700
Subject: [PATCH] TRITON-1365 server-setup job should not fail on first ETag
 conflict when marking server as setup Reviewed by: Orlando Vazquez
 <orlando@joyent.com> Approved by: Orlando Vazquez <orlando@joyent.com>

---
 lib/workflows/server-setup.js | 2 ++
 package.json                  | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/workflows/server-setup.js b/lib/workflows/server-setup.js
index f50e4f5..ca85e0a 100644
--- a/lib/workflows/server-setup.js
+++ b/lib/workflows/server-setup.js
@@ -60,6 +60,7 @@ function markServerAsSettingUp(job, callback) {
     var serverUrl = '/servers/' + job.params.server_uuid;
 
     var payload = {
+        etag_retries: 3,
         setting_up: true
     };
 
@@ -239,6 +240,7 @@ function markServerAsSetup(job, callback) {
     var serverUrl = '/servers/' + job.params.server_uuid;
 
     var payload = {
+        etag_retries: 3,
         setup: true,
         setting_up: false
     };
diff --git a/package.json b/package.json
index b7030e9..30ce103 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "SmartDataCenter Compute Node API",
-  "version": "1.22.0",
+  "version": "1.22.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
-- 
2.21.0

